### PR TITLE
Minor test-suite tweaks/fixes, volume V

### PR DESCRIPTION
**Pull request by dmnks**
These are mostly preparatory work for #2643.

### CODE REVIEW

File: tests/Dockerfile [hunk 1]
```diff
@@ -52,6 +52,23 @@ RUN dnf -y install \
   && dnf clean all
 RUN echo "%_dbpath $(rpm --eval '%_dbpath')" > /root/.rpmmacros
 
+# Workaround for pkgconf(1)'s unlisted dependency on rpm.
+# This is needed for cmake to work without an rpm installation.
+RUN ln -sf $(rpm --eval '%{_target_platform}%{?_gnu}')-pkg-config \
+           /usr/bin/pkg-config
+
+RUN rpm -e --nodeps --nodb \
```
Comment (pmatilai) [inline tests/Dockerfile:60]: Hey, somebody found a use-case for --nodb :partying_face:

File: tests/Dockerfile [hunk 2]
```diff
@@ -52,6 +52,23 @@ RUN dnf -y install \
   && dnf clean all
 RUN echo "%_dbpath $(rpm --eval '%_dbpath')" > /root/.rpmmacros
 
+# Workaround for pkgconf(1)'s unlisted dependency on rpm.
+# This is needed for cmake to work without an rpm installation.
+RUN ln -sf $(rpm --eval '%{_target_platform}%{?_gnu}')-pkg-config \
```
Comment (pmatilai) [inline tests/Dockerfile:57]: This is pretty sick really... (not objecting to what doing it, just grumbling as a reminder to myself to file a bug on Fedora's pkgconf)

File: tests/Dockerfile (hunk 1 above)
Comment (dmnks) [reply tests/Dockerfile:60]: Oh! Turns out this is a pretty new feature: 6de0d31ed3b4907dbd7e1683a18ea3a28790614d
So yup, good timing :laughing:

File: tests/Dockerfile (hunk 2 above)
Comment (dmnks) [reply tests/Dockerfile:57]: Yup, the funny thing is, I've also made myself a note back then to file a bug for this and of course never got around to it :smile:

**Pull request merged**